spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -499,13 +553,15 @@ public void testBinPackTableWithSpecialChars() {
   public void testSortTableWithSpecialChars() {
     Assume.assumeTrue(catalogName.equals(SparkCatalogConfig.HADOOP.catalogName()));
 
-    sql(
-        "CREATE TABLE %s (c1 int, c2 string, c3 string) USING iceberg",
-        tableName(QUOTED_SPECIAL_CHARS_TABLE_NAME));
+    String tblName = tableName(QUOTED_SPECIAL_CHARS_TABLE_NAME);
+    sql("CREATE TABLE %s (c1 int, c2 string, c3 string) USING iceberg", tblName);
 
-    insertData(tableName(QUOTED_SPECIAL_CHARS_TABLE_NAME), 10);
+    insertData(tblName, 10);
+    // TODO: metadata table access currently fails with special chars in the table name
+    // long dataSizeBefore = testDataSize(tblName);
```

Comment:
this seems like a bug to me where the table with special characters can't be found when running `SELECT sum(file_size_in_bytes) from %s.files`.

Reply:
turns out this happened because of caching in the Hadoop catalog. Adding `cache-enabled=false` fixes this issue